ntoskrnl/mm/balance.c

```diff
@@ -404,7 +406,14 @@ MiBalancerThread(PVOID Unused)
             while (InitialTarget != 0);
 
             if (Status == STATUS_WAIT_0)
-                InterlockedDecrement(&PageOutThreadActive);
+            {
+#if DBG
+                LONG Active = InterlockedExchange(&PageOutThreadActive, 0);
+                ASSERT(Active == 1);
+#else
+                InterlockedExchange(&PageOutThreadActive, 0);
+#endif
```

Comment:
```suggestion
                LONG Active = InterlockedExchange(&PageOutThreadActive, 0);
                ASSERT(Active == 1);
                DBG_UNREFERENCED_LOCAL_VARIABLE(Active);
```
(the `DBG_UNREFERENCED_LOCAL_VARIABLE` ensures that in non-DBG builds, where the ASSERT is compiled out, the unused Active variable doesn't trigger a compiler warning);

or:
```suggestion
                NT_VERIFY(InterlockedExchange(&PageOutThreadActive, 0) == 1);
```
See https://learn.microsoft.com/en-us/previous-versions/ff961903(v=vs.85)
and https://community.osr.com/t/nt-verify-does-not-work-as-expected/52250/5
for more details about `NT_VERIFY`.

Reply:
+1 for this correction of my suggestion.

Reply:
Done.